### ISSUE TITLE
io.mongock.driver.mongodb.springdata.v2.SpringDataMongoV2Driver required a bean of type 'long' that could not be found

### ISSUE THREAD

**Issue by nikhil-nomula** [flamingock/mongock#173]
## Description

io.mongock.driver.mongodb.springdata.v2.SpringDataMongoV2Driver required a bean of type 'long' that could not be found

## PRIORITY

NORMAL

### Mongock
- mongock-bom:5.0.29
- mongock-springboot:5.0.29, mongodb-springdata-v2-driver:5.0.29,
- annotation

### Environment
-  spring-boot:2.1.11.RELEASE spring-data-mongodb:2.1.11.RELEASE

## Steps to Reproduce

1. I include the above jars so that I can perform legacy migration from mongobee to mongock. 
2. When I run the app, I run into this error

```
2021-12-07 08:44:42.063  WARN 92627 --- [  restartedMain] ConfigServletWebServerApplicationContext : Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'getBuilder' defined in io.mongock.runner.springboot.config.MongockContext: Unsatisfied dependency expressed through method 'getBuilder' parameter 0; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'io.mongock.driver.mongodb.springdata.v2.SpringDataMongoV2Driver': Unsatisfied dependency expressed through constructor parameter 1; nested exception is org.springframework.beans.factory.NoSuchBeanDefinitionException: No qualifying bean of type 'long' available: expected at least 1 bean which qualifies as autowire candidate. Dependency annotations: {}
Related cause: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'getBuilder' defined in io.mongock.runner.springboot.config.MongockContext: Unsatisfied dependency expressed through method 'getBuilder' parameter 0; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'io.mongock.driver.mongodb.springdata.v2.SpringDataMongoV2Driver': Unsatisfied dependency expressed through constructor parameter 1; nested exception is org.springframework.beans.factory.NoSuchBeanDefinitionException: No qualifying bean of type 'long' available: expected at least 1 bean which qualifies as autowire candidate. Dependency annotations: {}
2021-12-07 08:44:42.082  INFO 92627 --- [  restartedMain] org.mongodb.driver.connection            : Closed connection [connectionId{localValue:4, serverValue:10}] to localhost:27017 because the pool has been closed.
2021-12-07 08:44:42.090  INFO 92627 --- [  restartedMain] org.mongodb.driver.connection            : Closed connection [connectionId{localValue:2, serverValue:8}] to localhost:27017 because the pool has been closed.
2021-12-07 08:44:42.094  INFO 92627 --- [  restartedMain] o.apache.catalina.core.StandardService   : Stopping service [Tomcat]
2021-12-07 08:44:42.107  INFO 92627 --- [  restartedMain] ConditionEvaluationReportLoggingListener : 

Error starting ApplicationContext. To display the conditions report re-run your application with 'debug' enabled.
2021-12-07 08:44:42.392 ERROR 92627 --- [  restartedMain] o.s.b.d.LoggingFailureAnalysisReporter   : 

***************************
APPLICATION FAILED TO START
***************************

Description:

Parameter 1 of constructor in io.mongock.driver.mongodb.springdata.v2.SpringDataMongoV2Driver required a bean of type 'long' that could not be found.


Action:

Consider defining a bean of type 'long' in your configuration.


Process finished with exit code 1


```

## Behaviour

**Expected behavior:** I am able to setup mongock and perform legacy migration

**Actual behavior:** Application does not start and runs into the error above

**How often the bug happens:** Happens all the time

## Additional context

application.yml looks like this

```
mongock:
  migration-scan-package:
    - com.org.changesets
  legacy-migration:
    origin: dbchangelog
    changes-count-expectation: 1
    run-always: false
    mapping-fields:
      change-id: changeId
      author: author
      timestamp: timestamp
      change-log-class: changeLogClass
      change-set-method: chnageSetMethod
```

**Comment by dieppa** [comment]
@nikhil-nomula  There was a bug in the Springdata v2 driver, nothing major, just injecting the wrong class to the context and the integration tests didn't spot that.

We have released version 5.0.30. It should fix it 😉 


On the other hand, you are using some older versions of Springboot and springdata, you will probably get some warnings around ilegal access. It's nothing to with Mongock, by upgrading your springbboot dependencies, this will vanish 

**please, let us know how it goes, so we can close this ticket"

**Comment by nikhil-nomula** [comment]
Thank you @dieppa , I updated to 5.0.30 and it now fails with this.. 


```
Error starting ApplicationContext. To display the conditions report re-run your application with 'debug' enabled.
2021-12-09 08:44:23.670 ERROR 45020 --- [  restartedMain] o.s.boot.SpringApplication               : Application run failed

org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'connectionDriver' defined in class path resource [io/mongock/driver/mongodb/springdata/v3/config/SpringDataMongoV3Context.class]: Unsatisfied dependency expressed through method 'connectionDriver' parameter 1; nested exception is org.springframework.beans.factory.NoSuchBeanDefinitionException: No qualifying bean of type '?' available: expected at least 1 bean which qualifies as autowire candidate. Dependency annotations: {}
	at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:800)
	at org.springframework.beans.factory.support.ConstructorResolver.instantiateUsingFactoryMethod(ConstructorResolver.java:541)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.instantiateUsingFactoryMethod(AbstractAutowireCapableBeanFactory.java:1352)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1195)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:582)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:542)
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:335)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:333)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:944)
	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:918)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:583)
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:145)
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:730)
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:412)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:302)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1301)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1290)
	at com.finishline.epic.zerodt.Application.main(Application.java:11)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.springframework.boot.devtools.restart.RestartLauncher.run(RestartLauncher.java:49)
Caused by: org.springframework.beans.factory.NoSuchBeanDefinitionException: No qualifying bean of type '?' available: expected at least 1 bean which qualifies as autowire candidate. Dependency annotations: {}
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.raiseNoMatchingBeanFound(DefaultListableBeanFactory.java:1790)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1346)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1300)
	at org.springframework.beans.factory.support.ConstructorResolver.resolveAutowiredArgument(ConstructorResolver.java:887)
	at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:791)
	... 24 common frames omitted

```
